moviepy/audio/io/AudioFileClip.py

```diff
@@ -74,6 +77,7 @@ def __init__(self, filename, buffersize=200000, nbytes=2, fps=44100):
         self.duration = self.reader.duration
         self.end = self.reader.duration
         self.buffersize = self.reader.buffersize
+        self.filename = filename
```

Comment:
IMHO we should, as a general rule, try to simplify the code as much as possible. this decorator is pretty simple but does the same than an explicit call to the function 

```suggestion
        self.filename = os.fspath(filename)
```

so, there is no really "logic" to encapsulate in the decorator. 

It's ok here, but my opinion we'll need be careful to don't abuse the "preprocessing" for stuff that are straight forward.

Reply:
Ok, that's a fair point. I suppose that my thinking was that using decorators it would be easy to see when a path-like object is allowed because it is right there above the function definition, instead of having to track down the `filename` variable and make sure that `os.fspath()` is applied to it before it used anywhere.